File: api/src/endpoint/runner/get_runner.rs
use crate::{endpoint::Endpoint, resource::RunnerResource};

pub struct GetRunner;

impl Endpoint for GetRunner {
    const PATH: &'static str = "/ci/runner/{owner}/{name}";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetRunnerRequest;
    type Response = GetRunnerResponse;
}

pub type GetRunnerRequest = ();
pub type GetRunnerResponse = RunnerResource;

File: gitdot-api-ts/src/endpoint/runner/get-runner.ts
import { z } from "zod";

import { RunnerResource } from "../../resource";

export const GetRunnerRequest = z.object({});
export type GetRunnerRequest = z.infer<typeof GetRunnerRequest>;

export const GetRunner = {
  path: "/ci/runner/{owner}/{name}",
  method: "GET",
  request: GetRunnerRequest,
  response: RunnerResource,
} as const;
export type GetRunner = typeof GetRunner;
